### PR TITLE
Mutant Dogs Initial Commit

### PR DESCRIPTION
**Pull request by RanikOrtega**
#### Summary
Content "Slightly mutated dog variants"

#### Purpose of change
To create extra content for "Pet-Keeper" playstyles and additional mutation options for dogs as they only had mongrels.

#### Describe the solution
Created some variations of dog mutations. Following suggestions from other users I kept mutations very mild as opposed to heavier mutations. Names subject to better suggestions: Loper = Dog with mutated legs, can jump short distances and climb objects. Mauler = Rottweiler with a wider and even more tooth filled mouth. Thrasher = small slower dogs that can dig.  

#### Describe alternatives you've considered
Not adding extra variants.

#### Testing
Spawned dogs in world with debug. Teleported to TCL-Lab finales and checked for dog variant spawns in Tier 2 mutagen labs.

#### Additional context
CDDA-Lab-Mutant

### CODE REVIEW

File: data/json/monsters/mutant_mammal.json [hunk 1]
```diff
@@ -369,5 +369,276 @@
       "SWARMS",
       "WARM"
     ]
+  },
+  {
+    "id": "mon_dog_mutant_loper",
+    "type": "MONSTER",
+    "name": { "str": "Loper" },
+    "description": "This dog has undergone visible mutation. While it's body is larger than normal it's legs are noticeably elongated, muscular and covered in long cord-like fur. It easily stands as tall as you and watches your every move closely with reflective eyes.",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:377]:
```suggestion
    "description": "This dog has undergone visible mutation.  While its body is larger than normal, its legs are noticeably elongated, muscular and covered in long cord-like fur.  It easily stands as tall as you and watches your every move closely with reflective eyes.",
```

File: data/json/monsters/mutant_mammal.json [hunk 2]
```diff
@@ -369,5 +369,276 @@
...
+      "CLIMBS"
+    ],
+    "armor": { "bash": 1, "cut": 1 }
+  },
+  {
+    "id": "mon_dog_mutant_loper_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog",
+    "name": { "str": "Loper puppy", "str_pl": "Loper puppies" },
+    "description": "An adorable, but unusual puppy. It wags it's tail happily but something about it's legs seems off. ",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:443]:
```suggestion
    "description": "An adorable, but unusual puppy.  It wags its tail happily but something about its legs seems off. ",
```

File: data/json/monsters/mutant_mammal.json [hunk 3]
```diff
@@ -369,5 +369,276 @@
...
+    "vision_night": 4,
+    "harvest": "dog_small_with_skull",
+    "upgrades": { "age_grow": 42, "into": "mon_dog_mutant_loper" },
+    "extend": { "flags": [ "NO_BREED" ] }
+  },
+  {
+    "id": "mon_dog_mutant_mauler",
+    "type": "MONSTER",
+    "name": { "str": "Mauler" },
+    "description": "This stocky dog has undergone visible mutation. It's jaw has widened and features more teeth than any dog should ever have. While it's toothy grin is unnerving, it's tail wags as energeticly as ever. ",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:465]:
```suggestion
    "description": "This stocky dog has undergone visible mutation.  Its jaw has widened and features more teeth than any dog should ever have.  While its toothy grin is unnerving, its tail wags as energetically as ever. ",
```

File: data/json/monsters/mutant_mammal.json [hunk 4]
```diff
@@ -369,5 +369,276 @@
...
+      "SWARMS",
+      "WARM"
+    ]
+  },
+  {
+    "id": "mon_dog_mutant_mauler_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog",
+    "name": { "str": "Mauler puppy", "str_pl": "Mauler puppies" },
+    "description": "An adorable, but unusual puppy. It wags it's tail happily but something about it's body seems off about it's mouth. ",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:534]:
```suggestion
    "description": "An adorable, but unusual puppy.  It wags its tail happily but something about its mouth seems off. ",
```

File: data/json/monsters/mutant_mammal.json [hunk 5]
```diff
@@ -369,5 +369,276 @@
...
+    "vision_night": 4,
+    "harvest": "dog_small_with_skull",
+    "upgrades": { "age_grow": 42, "into": "mon_dog_mutant_mauler" },
+    "extend": { "flags": [ "NO_BREED" ] }
+  },
+  {
+    "id": "mon_dog_mutant_thrasher",
+    "type": "MONSTER",
+    "name": { "str": "Thrasher" },
+    "description": "This dog has undergone visible mutation. It's paws have developed long digging nails and it's fur is patchily replaced with leathery skin. ",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:556]:
```suggestion
    "description": "This dog has undergone visible mutation.  Its paws have developed long digging nails and its fur is patchily replaced with leathery skin.",
```

File: data/json/monsters/mutant_mammal.json [hunk 6]
```diff
@@ -369,5 +369,276 @@
...
+      "WARM"
+    ],
+    "armor": { "bash": 1, "cut": 1 }
+  },
+  {
+    "id": "mon_dog_mutant_thrasher_pup",
+    "type": "MONSTER",
+    "copy-from": "mon_dog",
+    "name": { "str": "Thrasher puppy", "str_pl": "Thrasher puppies" },
+    "description": "An adorable, but unusual puppy. It wags it's tail happily but something about it's paws seems off. ",
```
Comment (TheShadowFerret) [inline data/json/monsters/mutant_mammal.json:626]:
```suggestion
    "description": "An adorable, but unusual puppy.  It wags its tail happily but something about its paws seems off. ",
```

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]
Its

File: data/json/monsters/mutant_mammal.json (hunk 1 above)
Comment (RanikOrtega) [reply data/json/monsters/mutant_mammal.json]: Changes updated in new commit.

**Comment by sadenar** [comment]
Might make snippets for them in a later PR, aside from that, good work.

**Comment by RanikOrtega** [comment]
Requesting review when possible. Thanks

**Comment by Maleclypse** [comment]
I'm not sure that these should be tameable as adults but with the other mutant mongrel being tameable I will go with it.

**Comment by RanikOrtega** [comment]
> I'm not sure that these should be tameable as adults but with the other mutant mongrel being tameable I will go with it.

These are the most normal of all mutant dogs. Essentially one or two things slightly off about them. Otherwise 90% dog. Anything more severe than these will be non-tameable unless paired with mods.

**Pull request merged**